api/src/org/apache/cloudstack/api/command/admin/host/ListHostsCmd.java

```diff
@@ -203,9 +201,22 @@ public void execute() {
                 hostResponse.setObjectName("host");
                 hostResponses.add(hostResponse);
             }
-
             response.setResponses(hostResponses, result.second());
         }
+        // Remove sensitive details from host response
+        List<HostResponse> hostResponsesList = response.getResponses();
+        for (HostResponse hostResponse: hostResponsesList) {
+            Map<String, String> hostDetails = hostResponse.getDetails();
+            if (hostDetails == null) continue;
+            if (hostDetails.containsKey("username")) {
+                hostDetails.remove("username");
+            }
+            if (hostDetails.containsKey("password")) {
+                hostDetails.remove("password");
+            }
+            hostResponse.setDetails(hostDetails);
```

Comment:
@bhaisaab I incorrectly assumed the HostResponse was only be used to create the ListHostCmd result to the client.  Based on the scope described, the location of this filtering is the narrowest scope possible.  We should avoid assumptions about how the ``HostDetails`` object is being consumed across threads, and make a defensive copy of it before modifying it.  Also, the code can be simplified by to remove the ``containsKey`` if blocks.